### PR TITLE
Laser Rifles!

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

Adds many more rechargable laser packs - AP weak, weak, disabler, disabler weak, pluse, xray, heavy
Adds in 2 new Rnd nodes - First one unlocks the laser rifles, weak laser shot, weak disabler shot
New disabler shot - a weaken verson that takes 12 shots to downsomeone!
Moves the carbine laser down into adv weapons as well as cost 5000 cm3 to make!

The gun itself can change out ammo types akin to the "Mag rifle" but are a laser type and EMPS cripple them much more
If a EMP hits a recharger pack, it dies. No fixing it, no recharging it, no more ammo inside it. Its dead Jim

## Why It's Good For The Game

Gives sec some more fire arms that are not all that amazing, but are not all that bad to give more options to carry around then raiding armory for blasters/carbines
As well as helps the power balance of the EMP.

## Changelog
:cl:
add: Old Laser rifles, and more ammo packs
balance: carbine laser blasters need plasma now
fix: Old Laser rifles not having a EMP act tho being a battery
/:cl:

### CODE REVIEW

File: code/modules/projectiles/boxes_magazines/external/rechargable.dm [hunk 1]
```diff
@@ -1,14 +1,68 @@
...
+/obj/item/ammo_box/magazine/recharge/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	else
+		max_ammo = 0
+		stored_ammo = 0
+		update_icon()
+		name = "fried power pack"
+		desc = "A once rechargeable, detachable cell, now fryed and useless. It has 0 shots left."
```
Comment (ghost) [inline code/modules/projectiles/boxes_magazines/external/rechargable.dm]: fried and useless, not fryed

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -646,19 +646,37 @@
...
-	design_ids = list("pin_loyalty")
+	prereq_ids = list("adv_engi", "laserrifle")
+	design_ids = list("pin_loyalty", "lasercarbine")
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 7500)
 	export_price = 5000
 
+/datum/techweb_node/laserrifle
+	id = "laserrifle"
+	display_name = "Out Dated Laser Weaponery"
+	description = "Looking back on older designs, will allow the furcher to not make the same misstakes."
```
Comment (ghost) [inline code/modules/research/techweb/all_nodes.dm]: grammatically there shouldn't be a comma here, and future not furcher

File: code/modules/research/techweb/all_nodes.dm [hunk 3]
```diff
@@ -646,19 +646,37 @@
...
+	prereq_ids = list("weaponry")
+	design_ids = list("laserrifle", "weak_power_pack", "power_pack_disabler_weak")
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
+	export_price = 5000
+
+/datum/techweb_node/improved_laserrifle
+	id = "improved_laserrifle"
+	display_name = "Refitted Laser Weaponery"
+	description = "Looking back on older designs, their still can be improvements made."
```
Comment (ghost) [inline code/modules/research/techweb/all_nodes.dm]: there not their

**Comment by silicons** [comment]
I don't like the perma-fry from EMP,

**Comment by Trilbyspaceclone** [comment]
How do you want emp's to work then on them?

**Comment by Arturlang** [comment]
Just make it empty itself in emp like everything else, maybe even the laser packs

**Comment by Ghommie** [comment]
Most laser firearms were designed to not be reloaded with new magazines on the go anyway, and we shouldn't really encourage people to go nuts with even more generic mag-feed firearms designs and their printable bottomless munition stockpiles.

**Comment by KathrinBailey** [comment]
Could we maybe get this reopened and at least have the weapon added to the game, even if not also added to the research list? Keeping it as admin spawn would be nice.

**Pull request closed without merging**